src/SerializerFactory.php

```diff
@@ -69,7 +69,8 @@ private function shouldUseObjectsForMaps() {
 	 * @return Serializer
 	 */
 	public function newEntitySerializer() {
-		$fingerprintSerializer = new FingerprintSerializer( $this->shouldUseObjectsForMaps() );
+		$options = $this->shouldUseObjectsForMaps() ? FingerprintSerializer::USE_OBJECTS_FOR_MAPS : FingerprintSerializer::USE_ARRAYS_FOR_MAPS;
```

Comment:
Imo we should make this a bitwise integer option like in `SerializerFactory` and we should also reuse the constants from there.